pkg/controller/kafkasecret/controller.go

```diff
@@ -39,11 +41,23 @@ func NewController(ctx context.Context, _ configmap.Watcher) *controller.Impl {
 		logger.Fatal("Failed To Load Environment Variables - Terminating!", zap.Error(err))
 	}
 
+	// Load the Sarama and other eventing-kafka settings from our configmap
+	// (though we don't need the Sarama settings here; the AdminClient loads them from the configmap each time it needs them)
+	_, configuration, err := commonconfig.LoadEventingKafkaSettings(ctx)
+	if err != nil {
+		logger.Fatal("Failed To Load Eventing-Kafka Settings", zap.Error(err))
+	}
+
+	// Overwrite some configmap settings with specific values provided by the environment
+	if err = env.VerifyOverrides(configuration, environment); err != nil {
+		logger.Fatal("Invalid / Missing Settings - Terminating", zap.Error(err))
+	}
+
 	// Create The KafkaSecret Reconciler
 	r := &Reconciler{
 		logger:             logging.FromContext(ctx),
 		kubeClientset:      kubeclient.Get(ctx),
-		environment:        environment,
+		config:             configuration,
 		kafkaChannelClient: injectionclient.Get(ctx),
```

Comment:
again - it's ok to have both - don't have to eliminate by mushing data into config - loses some clarity as to where stuff is coming from